pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java

```diff
@@ -240,8 +240,8 @@ && isBlank(starterArguments.bookieConfigFile)) {
 
             // init bookie server
             if (starterArguments.runBookie) {
-                checkNotNull(bookieConfig, "No ServerConfiguration for Bookie");
-                checkNotNull(bookieStatsProvider, "No Stats Provider for Bookie");
+                Objects.requireNonNull(bookieConfig, "No ServerConfiguration for Bookie");
```

Comment:
Why change `chekNotNull` to `Objects.requireNonNull` ?

Reply:
I think this decision is made in https://github.com/apache/pulsar/issues/12271 where we agree on enabling the Modernizer plugin and this is the default settings.

Besides, Guava gradually deprecates `Lists.newArrayList` since JDK provides it out-of-the-box. This could be the same to `chekNotNull` to `Objects.requireNonNull`.

Reply:
guava `chekNotNull` and jdk `Objects.requireNonNull` provide the same simple semantic:
```java
if (obj == null)
    throw new NullPointerException(message);
return obj;
```
@hangc0276